Update faiss to 1.12.0

xkong-anaconda opened the pull request:
faiss 1.12.0 

  **Destination channel:** Default

  ### Links

  - [PKG-10271](https://anaconda.atlassian.net/browse/PKG-10271)
  - [Upstream repository](https://github.com/facebookresearch/faiss)
  - [Upstream changelog/diff](https://github.com/facebookresearch/faiss/compare/v1.7.4...v1.12.0)
  - [Release notes v1.8.0](https://github.com/facebookresearch/faiss/releases/tag/v1.8.0)
  - [Release notes v1.9.0](https://github.com/facebookresearch/faiss/releases/tag/v1.9.0)
  - [Release notes v1.10.0](https://github.com/facebookresearch/faiss/releases/tag/v1.10.0)
  - [Release notes v1.11.0](https://github.com/facebookresearch/faiss/releases/tag/v1.11.0)
  - [Release notes v1.12.0](https://github.com/facebookresearch/faiss/releases/tag/v1.12.0)
  - Relevant dependency PRs:
    - None
    - 
### CUDA Variants
- [Linux CUDA Log](https://drive.google.com/file/d/1DsZ5Sjz5cIkRgldWIimrbE7PJAAwVWwm/view?usp=sharing)
- [Windows CUDA Log](https://drive.google.com/file/d/1ZlWRisEcCyvtGkSrkeg65G2WvDMG1sxm/view?usp=sharing)
  ### Explanation of changes:

  **Version update:** 1.7.4 → 1.12.0 

  **Updated recipe components:**
  - Version and SHA256 hash updated for v1.12.0 source
  - Added 6 new headers for v1.12.0 features:
    - `IndexIVFIndependentQuantizer.h` (v1.9.0)
    - `IndexIVFRaBitQ.h`, `IndexRaBitQ.h` (v1.11.0 - RaBitQ quantization)
    - `IndexNeuralNetCodec.h` (v1.9.0)
    - `gpu/GpuIndexBinaryCagra.h`, `gpu/GpuIndexCagra.h` (v1.12.0 - CAGRA GPU index)

  **Patches - Active (4):**
  All apply cleanly to v1.12.0 with expected line offsets:
  - `0003-skip-test_stress-for-GPU-on-windows.patch` 
  - `0004-enable-building-libfaiss_avx2-without-libfaiss.patch` 
  - `0005-increase-tolerance-for-marginally-failing-test.patch` 
  - `0006-add-bigobj-to-swigfaiss-compile-options-on-windows.patch` 

  **Patches - Removed (3):**
  Incorporated in upstream v1.12.0:
  - `0001-use-c-17.patch` - C++17 now default upstream
  - `0010-fix-index-type-in-openmp-loop.patch` - Fixed upstream
  - `0012-relax_test_ivf_train_2level.patch` - Fixed upstream

  **Patches - Commented out (5):**
  Pending CI verification:
  - `0002-adapt-header-target-directory-to-outputname.patch` - May be needed for AVX2 variant header installation
  - `0007-dont-use-GCC-extension-that-doesn-t-work-on-MSVC.patch` - Windows/MSVC only
  - `0008-ensure-_mm_prefetch-gets-char-not-float-as-first-arg.patch` - Windows/MSVC only
  - `0009-no-more-Release-subfolder.patch` - May be needed for Python packaging
  - `0011-add-missing-headers.patch` - Confirmed obsolete (headers present in v1.12.0)

  **Key upstream changes (v1.7.4 → v1.12.0):**
  - C++11 → C++17 
  - CUDA 10 dropped, CUDA 12 support added
  - Python 3.7 dropped
  - New features: AVX-512 optimizations, AMD ROCm/HIP support, ARM SVE support
  - New algorithms: RaBitQ quantization (v1.11.0), CAGRA GPU index (v1.12.0)
  - Optional: cuVS integration for NVIDIA GPUs (replaces RAFT)


- ...


[PKG-10271]: https://anaconda.atlassian.net/browse/PKG-10271?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

xkong-anaconda committed:
Version, SHA256, headers, patch references updated

xkong-anaconda committed:
Disable FAISS_ENABLE_EXTRAS to skip benchmarks/perf_tests

FAISS v1.12.0 added performance tests that require gflags dependency.
Since we don't need benchmarks/demos for the conda package, disable
them with -DFAISS_ENABLE_EXTRAS=OFF.

This fixes the build error:
  CMake Error: Could not find gflags package configuration

Fixes: PKG-10271

xkong-anaconda committed:
Fix: Set BUILD_TESTING=OFF to skip perf_tests

xkong-anaconda committed:
Use templated MKL pinnings to fix intel-openmp conflict

xkong-anaconda committed:
Fix faiss-proc skip issue

xkong-anaconda committed:
Remove empty requirements section

xkong-anaconda committed:
Add python runtime requirement

xkong-anaconda committed:
add host:python

xkong-anaconda committed:
force new faiss-proc artifacts

xkong-anaconda committed:
Added packaging to the requirements

xkong-anaconda committed:
Updated build-pkg.bat

xkong-anaconda committed:
skip external modele example test

package-build-platform commented:
<details><summary>Failed to submit graph for 8edf3e8c4c38512a3a006790ab0dfb34651faea0
</summary>Message:
Failed to run command: 'pbp generate from-feedstock --output /tmp/tmpum7npadr/faiss-split-feedstock.yaml --custom-metadata-file /tmp/tmpum7npadr/custom_metadata.yaml --release-name faiss-split-feedstock-2-8edf3e8c4c38512a3a006790ab0dfb34651faea0 faiss-split-feedstock/recipe', with error: 'INFO:root:Found feedstock faiss-split-feedstock at /tmp/tmpum7npadr/faiss-split-feedstock/recipe
INFO:root:Build/Test graph from faiss-split-feedstock
Created by sbx_user1051 on 2025-11-08 21:34:02
Command line: /opt/conda/envs/webhook-listener/bin/pbp generate from-feedstock --output /tmp/tmpum7npadr/faiss-split-feedstock.yaml --custom-metadata-file /tmp/tmpum7npadr/custom_metadata.yaml --release-name faiss-split-feedstock-2-8edf3e8c4c38512a3a006790ab0dfb34651faea0 faiss-split-feedstock/recipe

INFO:root:Found abs.yaml in /tmp/tmpum7npadr/faiss-split-feedstock
INFO:root:Creating graph from feedstock at: faiss-split-feedstock
INFO:pbpcli.utils.common:Setting env var: CONDA_SUBDIR = linux-64.
INFO:pbpcli.utils.expand_variants:Elapsed time for get_package_variants: 0.09 seconds
INFO:pbpcli.utils.expand_variants:Matrix vars: 2 unchanging vars:(397), changing vars (2): ['blas_impl', 'python']
INFO:pbpcli.utils.expand_variants:Elapsed time for preparation for render: 0.00 seconds
WARNING:urllib3.connectionpool:Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ReadTimeoutError("HTTPSConnectionPool(host='conda.anaconda.org', port=443): Read timed out. (read timeout=60.0)")': /sfe1ed40/linux-64/repodata.json
WARNING:urllib3.connectionpool:Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ReadTimeoutError("HTTPSConnectionPool(host='conda.anaconda.org', port=443): Read timed out. (read timeout=60.0)")': /sfe1ed40/linux-64/repodata.json
WARNING:urllib3.connectionpool:Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ReadTimeoutError("HTTPSConnectionPool(host='conda.anaconda.org', port=443): Read timed out. (read timeout=60.0)")': /sfe1ed40/linux-64/repodata.json
WARNING:urllib3.connectionpool:Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ReadTimeoutError("HTTPSConnectionPool(host='conda.anaconda.org', port=443): Read timed out. (read timeout=60.0)")': /sfe1ed40/linux-64/repodata.json
INFO:pbpcli.utils.expand_variants:Elapsed time for api.render: 403.27 seconds
INFO:pbpcli.utils.common:Unsetting env var: CONDA_SUBDIR
INFO:pbpcli.utils.common:Setting env var: CONDA_SUBDIR = linux-aarch64.
INFO:pbpcli.utils.expand_variants:Elapsed time for get_package_variants: 0.09 seconds
INFO:pbpcli.utils.expand_variants:Matrix vars: 1 unchanging vars:(397), changing vars (1): ['python']
INFO:pbpcli.utils.expand_variants:Elapsed time for preparation for render: 0.00 seconds
INFO:pbpcli.utils.common:Setting env var: CONDA_OVERRIDE_GLIBC = 2.28.
WARNING: Number of parsed outputs does not match detected raw metadata blocks. Identified output block may be wrong! If you are using Jinja conditionals to include or exclude outputs, consider using `skip: true  # [condition]` instead.
WARNING:urllib3.connectionpool:Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ReadTimeoutError("HTTPSConnectionPool(host='conda.anaconda.org', port=443): Read timed out. (read timeout=60.0)")': /sfe1ed40/linux-aarch64/repodata.json
WARNING:urllib3.connectionpool:Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ReadTimeoutError("HTTPSConnectionPool(host='conda.anaconda.org', port=443): Read timed out. (read timeout=60.0)")': /sfe1ed40/linux-aarch64/repodata.json
WARNING:urllib3.connectionpool:Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ReadTimeoutError("HTTPSConnectionPool(host='conda.anaconda.org', port=443): Read timed out. (read timeout=60.0)")': /sfe1ed40/linux-aarch64/repodata.json
WARNING: failed to get package records, retrying.  exception was: HTTP 000 CONNECTION FAILED for url <https://conda.anaconda.org/sfe1ed40/linux-aarch64/repodata.json>
Elapsed: -

An HTTP error occurred when trying to retrieve this URL.
HTTP errors are often intermittent, and a simple retry will get you on your way.
'https//conda.anaconda.org/sfe1ed40/linux-aarch64'

WARNING:urllib3.connectionpool:Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ReadTimeoutError("HTTPSConnectionPool(host='conda.anaconda.org', port=443): Read timed out. (read timeout=60.0)")': /sfe1ed40/linux-aarch64/repodata.json
INFO:pbpcli.utils.common:Unsetting env var: CONDA_OVERRIDE_GLIBC
INFO:pbpcli.utils.common:Unsetting env var: CONDA_SUBDIR
Traceback (most recent call last):
  File "/opt/conda/envs/webhook-listener/bin/pbp", line 10, in <module>
    sys.exit(cli())
             ~~~^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1442, in __call__
    return self.main(*args, **kwargs)
           ~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1363, in main
    rv = self.invoke(ctx)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1830, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1830, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1226, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 794, in invoke
    return callback(*args, **kwargs)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/cli/generate.py", line 413, in wrapper
    return func(*args, **kwargs)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/cli/generate.py", line 624, in from_feedstock
    build_configs, test_configs = expand_build_matrix(feedstock.recipe_dir_path, src_channels)
                                  ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/cli/generate.py", line 79, in expand_build_matrix
    _targets, build_configs, test_configs = variants_for_feedstock_and_platform(
                                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        recipe_dir_path, platform, src_channels
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/utils/expand_variants.py", line 544, in variants_for_feedstock_and_platform
    metadata_tuples = api.render(
        parsed.recipe,
    ...<2 lines>...
        variants=None,
    )
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/api.py", line 68, in render
    return render_metadata_tuples(
        metadata_tuples,
    ...<3 lines>...
        bypass_env_check=bypass_env_check,
    )
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/render.py", line 1097, in render_metadata_tuples
    for od, om in meta.get_output_metadata_set(
                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        permit_unsatisfiable_variants=permit_unsatisfiable_variants,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        permit_undefined_jinja=not finalize,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        bypass_env_check=bypass_env_check,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ):
    ^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/metadata.py", line 2781, in get_output_metadata_set
    conda_packages = finalize_outputs_pass(
        ref_metadata,
    ...<3 lines>...
        bypass_env_check=bypass_env_check,
    )
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/metadata.py", line 1077, in finalize_outputs_pass
    fm = finalize_metadata(
        om,
        parent_metadata=parent_metadata,
        permit_unsatisfiable_variants=permit_unsatisfiable_variants,
    )
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/render.py", line 722, in finalize_metadata
    build_unsat, host_unsat = add_upstream_pins(
                              ~~~~~~~~~~~~~~~~~^
        m, permit_unsatisfiable_variants, exclude_pattern, extra_specs
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/render.py", line 507, in add_upstream_pins
    host_deps, host_unsat, extra_run_specs_from_host = _read_upstream_pin_files(
                                                       ~~~~~~~~~~~~~~~~~~~~~~~~^
        m, "host", permit_unsatisfiable_variants, exclude_pattern, extra_specs
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/render.py", line 463, in _read_upstream_pin_files
    extra_run_specs = get_upstream_pins(m, precs, env)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/render.py", line 433, in get_upstream_pins
    loc, dist = execute_download_actions(
                ~~~~~~~~~~~~~~~~~~~~~~~~^
        m,
        ^^
    ...<2 lines>...
        package_subset=[prec],
        ^^^^^^^^^^^^^^^^^^^^^^
    )[prec]
    ^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/render.py", line 388, in execute_download_actions
    link_prec = [
                ~
    ...<3 lines>...
        == (prec.name, prec.version, prec.build)
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    ][0]
    ~^^^
IndexError: list index out of range
'
aws lambda request id: a8301b3e-ae9e-525b-bf8a-7ab7171f1b21
[View in Grafana](https://grafana.anacondaconnect.com/explore?schemaVersion=1&panes=%7B%2286g%22:%7B%22datasource%22:%22loki-central%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22%7B__aws_cloudwatch_log_group%3D%5C%22%2Faws%2Flambda%2Fpbp-graph-submitter-prod%5C%22%7D%20%7C%20json%20%7C%20requestId%20%3D%20%60a8301b3e-ae9e-525b-bf8a-7ab7171f1b21%60%22,%22queryType%22:%22range%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22loki-central%22%7D,%22editorMode%22:%22builder%22,%22direction%22:%22backward%22%7D%5D,%22range%22:%7B%22from%22:%22now-3h%22,%22to%22:%22now%22%7D,%22panelsState%22:%7B%22logs%22:%7B%22columns%22:%7B%220%22:%22Time%22,%221%22:%22Line%22%7D,%22visualisationType%22:%22table%22,%22labelFieldName%22:%22labels%22%7D%7D%7D%7D&orgId=1)</details>

xkong-anaconda committed:
trigger CI

xkong-anaconda committed:
Updated both test runners to remove the offending file before running pytest

xkong-anaconda committed:
Added a new source patch

xkong-anaconda committed:
patch update

xkong-anaconda committed:
patch update

xkong-anaconda committed:
Build vendored llama.cpp instead of using external dependency

xkong-anaconda committed:
skipping python3.13 for osx

xkong-anaconda committed:
updated recipe/build-lib.sh to pass -DBLA_VENDOR=OpenBLAS

xkong-anaconda committed:
force CMake to use the OpenBLAS

xkong-anaconda committed:
revents CMake from probing for separate LAPACK libs

xkong-anaconda committed:
set env

xkong-anaconda committed:
test fix

xkong-anaconda committed:
fix for win tests

xkong-anaconda committed:
Skip Generic (non-AVX2) test on Windows

xkong-anaconda committed:
regenerate patch

xkong-anaconda committed:
fix for gpu header

tferguson47 commented:
@xkong-anaconda  This should be going to the main channel, not snowflake

cbouss commented on recipe/conda_build_config.yaml:
```diff
@@ -2,3 +2,10 @@ cxx_compiler:          # [win]
   - vs2019               # [win]
 cuda_compiler_version:
```
We do want to build cuda variants.

xkong-anaconda replied:
CUDA variants are build:
- [Linux CUDA Log](https://drive.google.com/file/d/1DsZ5Sjz5cIkRgldWIimrbE7PJAAwVWwm/view?usp=sharing)
- [Windows CUDA Log](https://drive.google.com/file/d/1ZlWRisEcCyvtGkSrkeg65G2WvDMG1sxm/view?usp=sharing)

xkong-anaconda committed:
add cuda

xkong-anaconda committed:
Add CUDA host dependencies for GPU builds

xkong-anaconda committed:
Add CUDA build gating to prevent automatic CI builds

xkong-anaconda committed:
update to cuda 12.8

cbouss commented on recipe/meta.yaml:
```diff
@@ -255,21 +267,22 @@ outputs:
         # MKL for win, linux-64, and osx-64. All require intel-openmp.
-        - mkl 2023.1.0            # [not (s390x or ppc64le or aarch64 or arm64)] 
+        - mkl {{ mkl }}           # [not (s390x or ppc64le or aarch64 or arm64)]
         # mkl-devel seems to be required for win.
```
```suggestion
```

cbouss commented on recipe/meta.yaml:
```diff
@@ -255,21 +267,22 @@ outputs:
+        - mkl {{ mkl }}           # [not (s390x or ppc64le or aarch64 or arm64)]
         # mkl-devel seems to be required for win.
-        - mkl-devel 2023.1.0      # [win]
+        - mkl-devel {{ mkl }}     # [win]
```
```suggestion
        - mkl-devel {{ mkl }}     # [blas_impl == "mkl"]
```

cbouss commented on recipe/meta.yaml:
```diff
@@ -255,21 +267,22 @@ outputs:
-        # OpenBLAS + libgomp for osx-arm64, and linux-s390x and linux-ppc64le, and linux-aarch64. 
-        - openblas 0.3.21         # [s390x or ppc64le or aarch64 or arm64]     
+        # OpenBLAS + libgomp for osx-arm64, and linux-s390x and linux-ppc64le, and linux-aarch64.
+        - openblas 0.3.21         # [s390x or ppc64le or aarch64 or arm64]
```
```suggestion
        - openblas-devel {{ openblas }}  # [blas_impl == "openblas"]
```

cbouss commented on recipe/meta.yaml:
```diff
@@ -255,21 +267,22 @@ outputs:
         - libfaiss-avx2 ={{ version }}=*_{{ faiss_proc_type }}  # [x86_64]
         - {{ pin_compatible('numpy') }}
+        - packaging
         - libopenblas            # [s390x or ppc64le or aarch64 or arm64]
```
```suggestion
        # bounds through run_exports
        - libopenblas            # [blas_impl == "openblas"]
```

cbouss commented on recipe/meta.yaml:
```diff
@@ -255,21 +267,22 @@ outputs:
+        - packaging
         - libopenblas            # [s390x or ppc64le or aarch64 or arm64]
-        - mkl 2023.1.*           # [not (s390x or ppc64le or aarch64 or arm64)]
+        - mkl {{ mkl }}.*        # [not (s390x or ppc64le or aarch64 or arm64)]
```
```suggestion
        - mkl                # [blas_impl == "openblas"]
```

cbouss commented on recipe/meta.yaml:
Might want to update pin to a version matching the llvm being pulled

cbouss commented on recipe/meta.yaml:
Probably not needed.

cbouss commented on recipe/meta.yaml:
```diff
@@ -255,21 +267,22 @@ outputs:
         - python
         - libfaiss ={{ version }}=*_{{ faiss_proc_type }}
         - libfaiss-avx2 ={{ version }}=*_{{ faiss_proc_type }}  # [x86_64]
         - {{ pin_compatible('numpy') }}
```
```suggestion
        - numpy
```
Bounds come from run_exports, and yield a higher range than pin_compatible in this case.

cbouss commented on recipe/meta.yaml:
```diff
@@ -131,21 +135,23 @@ requirements:
     - intel-openmp            # [linux64 or win]
     # mkl-devel seems to be required for win.
-    - mkl-devel 2023.1.0      # [win]
+    - mkl-devel {{ mkl }}     # [win]
```
Review blas pinnings

xkong-anaconda replied:
BLAS pinnings reviewed and  blas_impl selectors are updated as suggested

cbouss commented on recipe/meta.yaml:
```diff
@@ -131,21 +135,23 @@ requirements:
+      host:
+        - python
+      run:
+        - python
```
```suggestion
```
Not needed

cbouss commented on recipe/meta.yaml:
```diff
@@ -131,21 +135,23 @@ requirements:
-      skip: true  # [py==38 and osx and x86_64]
-      string: {{ faiss_proc_type }}
+      # include hash+buildnum so --skip-existing does not skip refreshed builds
+      string: "h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}_{{ faiss_proc_type }}"
```
We should keep string as {{ faiss_proc_type }}

See example: https://github.com/AnacondaRecipes/openblas-feedstock/blob/master/recipe/meta.yaml#L127-L153

xkong-anaconda replied:
The string variable for faiss-proc has been updated

xkong-anaconda replied:
After examining it closely, Iit actually supports using a hash in the faiss-proc build string.
the openblas-feedstock lines 127-130, the blas package with simple string openblas is commented out with this note:
  "Since this package has a non-unique name, blas-1.0-openblas.conda, it is omitted from future 
  updates as they cause issues with overwrites on the PBP. If an update to this package is 
  needed in the future, the name should have a hash added so that it becomes unique per 
  publish."

The active openblas meta-package (lines ~117-125) has no explicit build string, meaning it uses conda's default hash-based string - not a simple variant string. So simple strings like openblas or cpu may cause "issues with overwrites on the PBP". So i still keep the hash for this case

cbouss commented on recipe/conda_build_config.yaml:
```diff
@@ -1,4 +1,17 @@
+  - "3.12"
+  - "3.13"           # [not osx]
+  # Python 3.13 builds currently fail on macOS arm64 (upstream SWIG bindings not ready),
+  # so keep it enabled elsewhere but skip on osx for now.
```
Please provide an upstream link as a comment if you have it.

xkong-anaconda replied:
The upstream link for the Python 3.13 build failures on macOS ARM64 has been added as a comment

xkong-anaconda committed:
Update recipe/meta.yaml

Co-authored-by: Charles Bousseau <16641587+cbouss@users.noreply.github.com>

xkong-anaconda committed:
Update recipe/meta.yaml

Co-authored-by: Charles Bousseau <16641587+cbouss@users.noreply.github.com>

xkong-anaconda committed:
Update recipe/meta.yaml

Co-authored-by: Charles Bousseau <16641587+cbouss@users.noreply.github.com>

xkong-anaconda committed:
Update recipe/meta.yaml

Co-authored-by: Charles Bousseau <16641587+cbouss@users.noreply.github.com>

xkong-anaconda committed:
Update recipe/meta.yaml

Co-authored-by: Charles Bousseau <16641587+cbouss@users.noreply.github.com>

xkong-anaconda committed:
Update recipe/meta.yaml

Co-authored-by: Charles Bousseau <16641587+cbouss@users.noreply.github.com>

xkong-anaconda committed:
Update recipe/meta.yaml

Co-authored-by: Charles Bousseau <16641587+cbouss@users.noreply.github.com>

xkong-anaconda committed:
Added the blas_impl configuration

xkong-anaconda committed:
update string variable for faiss-proc

xkong-anaconda committed:
Add python back to the host requirements of the faiss-proc output in recipe/meta.yaml.

  This restores the link between the faiss-proc output and the build matrix, ensuring it gets built alongside the other outputs.

xkong-anaconda committed:
keep hash

xkong-anaconda committed:
Added c_stdlib and c_stdlib_version - Required for proper build configuration

xkong-anaconda committed:
New patch: 0014-fix-MSVC-template-disambiguation.patch

xkong-anaconda committed:
Added CUDA dev packages

xkong-anaconda committed:
add headers

xkong-anaconda committed:
removed python from the libfaiss host requirements

ELundby45 commented on recipe/conda_build_config.yaml:
```diff
@@ -1,4 +1,39 @@
 cxx_compiler:          # [win]
   - vs2019               # [win]
+
```
The majority of what's in this conda_build_config.yaml is already in the main CBC and has the same values, do we need to re-define everything here?

xkong-anaconda replied:
Good catch. Removed the duplicate entries

ELundby45 commented on recipe/meta.yaml:
```diff
@@ -189,19 +195,25 @@ outputs:
         # llvm-openmp is required only for osx
         - llvm-openmp  14.0.6     # [osx]
```
Would we want to just match the same version of llvm-openmp as the compiler we are using?


```suggestion
        - llvm-openmp  {{ cxx_compiler_version }}     # [osx]
```

xkong-anaconda replied:
Good suggestion. Updated the build requirement .

ELundby45 commented on recipe/meta.yaml:
```suggestion
        - llvm-openmp  {{ cxx_compiler_version }}     # [osx]
```
Same as above.

ELundby45 reviewed (changes requested):
The abs.yaml file needs to be deleted.

xkong-anaconda committed:
Remove duplicate CBC entries inherited from main

xkong-anaconda committed:
Address reviewer comments on lvm-openmp

xkong-anaconda committed:
Address reviewer feedback

   Co-Authored-By: Eric Lundby <elundby@anaconda.com>

howea commented on recipe/meta.yaml:
```diff
@@ -40,6 +40,7 @@
 ] + (not win) * [
     'invlists/OnDiskInvertedLists.h'
 ] + (cuda_compiler_version != "None") * [
+    # Note: GpuIndexBinaryCagra.h and GpuIndexCagra.h are only installed with FAISS_ENABLE_CUVS=ON
```
This line may be causing the linter error. I suspect jinja (or the linter) is getting confused by the comment inside the jinja markup.
```
===== ERRORS =====
- /tmp/tmpbupw63d4/faiss-split-feedstock/recipe/:43: jinja_render_failure: The recipe could not be rendered by Jinja2```

howea replied:
Ok, I saw your update. I was not expecting that change to cause a further 22,000 linter errors. I'm not sure what is causing those extra errors

ELundby45 replied:
I wouldn't stress about trying to solve the linter. It's never been able to handle complex meta.yaml yaml's such as this.

xkong-anaconda committed:
Remove comment inside Jinja expression to fix linter error

JamesRobertsonGames commented on recipe/conda_build_config.yaml:
```diff
@@ -1,4 +1,24 @@
 cuda_compiler_version:
-  - None
+  - None                                                                       # CPU variant - always built
+  - 12.8               # [ANACONDA_ROCKET_ENABLE_CUDA and (linux64 or win)]  # GPU variant - manual only
```
should be able to just do this now 
```suggestion
  - 12.8               # [win or (linux and x86_64)]
```

JamesRobertsonGames commented on recipe/conda_build_config.yaml:
```diff
@@ -1,4 +1,24 @@
+
+blas_impl:
+  - mkl              # [not (aarch64 or ppc64le or s390x or arm64)]
+  - openblas         # [aarch64 or ppc64le or s390x or arm64]
```
out of date platforms 
```suggestion
  - mkl              # [not (aarch64  or arm64)]
  - openblas         # [aarch64 or arm64]
```

JamesRobertsonGames commented on recipe/build-lib.sh:
```diff
@@ -5,6 +5,27 @@ cd ${SRC_DIR}
+# not attempt to run detection binaries on the build host (which fails when
+# cross compiling). Also hand CMake the actual library path so it does not try
+# to execute detection binaries under emulation.
+if [[ "${target_platform}" == linux-aarch64 || "${target_platform}" == linux-ppc64le || "${target_platform}" == linux-s390x || "${target_platform}" == osx-arm64 ]]; then
```
```suggestion
if [[ "${target_platform}" == linux-aarch64 || "${target_platform}" == osx-arm64 ]]; then
```

xkong-anaconda replied:
Thank you for the suggestion

JamesRobertsonGames reviewed (commented):
some minor adjustments - agree with Charles' comments that are still in play

Should be able to run on CI now too

xkong-anaconda committed:
Update recipe/conda_build_config.yaml

Co-authored-by: James Robertson <jamierobertsongames@gmail.com>

xkong-anaconda committed:
Update recipe/conda_build_config.yaml

Co-authored-by: James Robertson <jamierobertsongames@gmail.com>

xkong-anaconda committed:
Update recipe/build-lib.sh

Co-authored-by: James Robertson <jamierobertsongames@gmail.com>

xkong-anaconda committed:
Fix FAISS_DISABLE_CPU_FEATURES env var parsing bug

xkong-anaconda committed:
patch fix

xkong-anaconda commented:
> some minor adjustments - agree with Charles' comments that are still in play
> 
> Should be able to run on CI now too

Thank you. The change has been made
